### ISSUE TITLE
Blob too large: no documented way to chunk uploads

### ISSUE THREAD

**Issue by sanmai-NL** [pulumi/pulumi-azure-native#2150]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

I can't find documentation on how to upload a large file (of about 1.2 GiB) using Pulumi's `FileAsset` class. It's possible using the Azure's own Python SDK.

```text
  azure-native:storage:Blob (vhd-linux):
    error: blobs.Client#PutBlockBlob: Failure responding to request: StatusCode=413 -- Original Error: autorest/azure: Service returned an error. Status=413 Code="RequestBodyTooLarge" Message="The request body is too large and exceeds the maximum permissible limit.\nRequestId:39dda59b-701e-0061-083a-155045000000\nTime:2022-12-21T12:51:17.7652469Z"
```

I am using Pulumi v3.50.0.

**Comment by danielrbradley** [comment]
Thanks for writing this up @sanmai-NL 

## Workarounds

As mentioned, this is possible via the Azure SDK. You can also wrap the SDK in a dynamic provider to use it within the lifecycle of a Pulumi program: https://github.com/pulumi/examples/tree/master/aws-py-dynamicresource

Another approach could be to use the [Pulumi Command Provider](https://www.pulumi.com/registry/packages/command/) to do this operation via the Azure CLI.

**Comment by sanmai-NL** [comment]
Understood. Thanks for taking interest in this. From a user perspective, though, the whole idea of infrastructure-as-code with Puluki falls apart when you have to shell out to some executable in the middle of a deployment process, with the auth and reproducibility/predictability problems that come with it.

Side note, Microsoft seems to advise [`azcopy`](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-blobs-upload) for these operations. Perhaps Pulumi can wrap that.

**Comment by danielrbradley** [comment]
Yup, would be good to be able to solve this properly in the provider. The workaround is just the temporary stop-gap until we can address the issue at its root.
